dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/SetShippingMethodsOnCartTest.php

```diff
@@ -42,100 +48,437 @@ protected function setUp()
+
+        $maskedQuoteId = $this->getMaskedQuoteIdByReversedQuoteId($reservedOrderId);
+        $quoteAddressId = $this->getQuoteAddressIdByReversedQuoteId($reservedOrderId);
+
+        $query = $this->prepareMutationQuery(
+            $maskedQuoteId,
+            $methodCode,
+            $carrierCode,
+            $quoteAddressId
+        );
+
+        $this->graphQlQuery($query);
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Sales/_files/guest_quote_with_addresses.php
+     */
     public function testSetNonExistentShippingMethod()
     {
-        $this->markTestIncomplete('https://github.com/magento/graphql-ce/issues/422');
+        $methodCode = 'non-existed-method-code';
+        $carrierCode = 'non-carrier-method-code';
+        $reservedOrderId = 'guest_quote';
+
+        $maskedQuoteId = $this->getMaskedQuoteIdByReversedQuoteId($reservedOrderId);
+
+        /** @var Quote $quote */
+        $quote = $this->quoteFactory->create();
+        $this->quoteResource->load($quote, 'guest_quote', 'reserved_order_id');
+        $quoteAddressId = (int) $quote->getShippingAddress()->getId();
+
+        $query = $this->prepareMutationQuery(
+            $maskedQuoteId,
+            $methodCode,
+            $carrierCode,
+            $quoteAddressId
+        );
+
+        self::expectException(\Exception::class);
+        self::expectExceptionMessage('Carrier with such method not found: ' . $carrierCode . ', ' . $methodCode . '');
+        $this->graphQlQuery($query);
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Sales/_files/guest_quote_with_addresses.php
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_virtual_product_and_address.php
+     */
     public function testSetShippingMethodIfAddressIsNotBelongToCart()
     {
-        $this->markTestIncomplete('https://github.com/magento/graphql-ce/issues/422');
+        $methodCode = 'flatrate';
+        $carrierCode = 'flatrate';
+        $reservedOrderId = 'guest_quote';
+
+        $maskedQuoteId = $this->getMaskedQuoteIdByReversedQuoteId($reservedOrderId);
+
+        /** @var Quote $quote */
+        $quote = $this->quoteFactory->create();
+        $this->quoteResource->load($quote, 'test_order_with_virtual_product', 'reserved_order_id');
+        $quoteAddressId = (int) $quote->getShippingAddress()->getId();
+
+        $query = $this->prepareMutationQuery(
+            $maskedQuoteId,
+            $methodCode,
+            $carrierCode,
+            $quoteAddressId
+        );
+
+        self::expectException(\Exception::class);
+        self::expectExceptionMessage('The current user cannot use cart address with ID "' . $quoteAddressId . '"');
+        $this->graphQlQuery($query);
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Sales/_files/guest_quote_with_addresses.php
+     * @expectedException \Magento\Framework\Exception\NoSuchEntityException
+     * @expectedExceptionMessage No such entity with cartId = 0
```

Comment:
According to implementation message should be 
`Could not find a cart with ID "non_existent_masked_id"`

Reply:
Changed the message and updated test case based on it.